router/batchrouter/batchrouter.go

```diff
@@ -897,24 +901,32 @@ func (worker *workerT) workerProcess() {
 						}
 
 						destUploadStat.End()
+						// td: add case foor s3 anmd batch here based on time window - batch opn receivedAt
 					case misc.ContainsString(warehouseDestinations, brt.destType):
 						useRudderStorage := misc.IsConfiguredToUseRudderObjectStorage(batchJobs.BatchDestination.Destination.Config)
 						objectStorageType := warehouseutils.ObjectStorageType(brt.destType, batchJobs.BatchDestination.Destination.Config, useRudderStorage)
 						destUploadStat := stats.NewStat(fmt.Sprintf(`batch_router.%s_%s_dest_upload_time`, brt.destType, objectStorageType), stats.TimerType)
 						destUploadStat.Start()
-						output := brt.copyJobsToStorage(objectStorageType, batchJobs, true, true)
-						postToWarehouseErr := false
-						if output.Error == nil && output.Key != "" {
-							output.Error = brt.postToWarehouse(batchJobs, output)
-							if output.Error != nil {
-								postToWarehouseErr = true
+						splitBatchJobs := brt.SplitBatchJobsOnTimeWindow(batchJobs)
+						// td: add new case for s3
+						// td: move s3 specific stuff to s3 case
+						for _, batchJobPtr := range splitBatchJobs {
+							batchJob := *batchJobPtr
+							output := brt.copyJobsToStorage(objectStorageType, batchJob, true, true)
```

Comment:
let's change `copyJobsToStorage` to take pointer for batchJob instead of value just to not take up extra mem ?